.github/workflows/spec.yaml

```diff
@@ -1,11 +1,22 @@
-on: [ push ]
+name: Test and build
+on: push
 
 jobs:
-  spec:
-    name: RSpec
+  build:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - uses: actions/setup-ruby@v1
-    - run: bundle install
+    - id: branch_name
+      run: echo ::set-output name=branch::${GITHUB_REF#refs/heads/}
+    - uses: ruby/setup-ruby@v1
+      with:
+        bundler-cache: true
     - run: bundle exec rspec
+    - uses: docker/build-push-action@v1
```

Comment:
Why are pushing an image after running specs?

Reply:
It's useful for testing. It gets tagged to the branch. Is there a reason we may not wish to do this?

Reply:
Is there some sort of API limit to the amount of images we can push to docker? 

I wouldn't expect it on a spec run, but if it useful 👍

Reply:
Nope, and since we're tagging it with the same tag, it'll replace the image in Docker Hub.